MAUI/DataGrid/sorting.md

```diff
@@ -0,0 +1,252 @@
+    {
+        e.Cancel = true;
+    }
+}
+{% endhighlight %}
+
+{% highlight c# %}
+this.sfDataGrid.SortColumnsChanging += SfDataGrid_SortColumnsChanging;
+
+private void sfDataGrid_SortColumnsChanging(object sender, DataGridSortColumnsChangingEventArgs e)
+{
+    if (e.AddedItems[0].ColumnName == "OrderID")
+    {
+        e.Cancel = true;
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+## Disable sorting for an individual column
+
+The `SfDataGrid` disables sorting for an individual column by setting the [DataGridColumn.AllowSorting](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridColumn.html#Syncfusion_Maui_DataGrid_DataGridColumn_AllowSorting) property to false. The default value of this property is true.
+
+### for auto generated columns
+
+{% tabs %}
+{% highlight xaml %}
+<syncfusion:SfDataGrid x:Name="sfDataGrid" AutoGeneratingColumn="sfDataGrid_AutoGeneratingColumn" SortingMode="Single" ItemsSource="{Binding OrderInfoCollection}" >
+</syncfusion:SfDataGrid>
+
+private void sfDataGrid_AutoGeneratingColumn(object sender, DataGridAutoGeneratingColumnEventArgse)
+{
+    if (e.Column.MappingName == "OrderID")
+    {
+        e.Column.AllowSorting = false;
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+### for manually defined columns
+
+{% tabs %}
+{% highlight xaml %}
+<syncfusion:SfDataGrid x:Name="sfDataGrid" AutoGenerateColumnsMode="None" SortingMode="Single"ItemsSource="{Binding OrderInfoCollection}" >
+    <syncfusion:SfDataGrid.Columns>
+        <syncfusion:DataGridNumericColumn HeaderText="Order ID" AllowSorting="False" MappingName="OrderID" />
+        <syncfusion:DataGridTextColumn HeaderText="Customer ID" MappingName="CustomerID" />
+        <syncfusion:DataGridTextColumn HeaderText="Customer" MappingName="Customer" />
+        <syncfusion:DataGridTextColumn HeaderText="Ship City" MappingName="ShipCity" />
+        <syncfusion:DataGridTextColumn HeaderText="Ship Country" MappingName="ShipCountry" />
+    </syncfusion:SfDataGrid.Columns>
+</syncfusion:SfDataGrid>
+{% endhighlight %}
+{% endtabs %}
+
+## Custom sorting
+
+The `SfDataGrid` sort columns based on custom logic, when the standard sorting techniques do not meet the requirements. For each column, you can apply different sorting criteria by adding [SortComparer](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Data.SortComparer.html) objects to [SfDataGrid.SortComparers](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_SortComparers) collection.
+
+The `SortComparer` object contains the following properties,
+
+* **PropertyName**: `MappingName` of the column that applies custom sorting.
+* **Comparer**: Gets or sets the custom comparer that implements the [IComparer](https://docs.microsoft.com/en-us/dotnet/api/system.collections.icomparer?view=net-6.0) and [ISortDirection](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Data.ISortDirection.html) interfaces.
+
+{% tabs %}

```

Comment:
@ashok-kuvaraja  - Add the details,

The following example shows how to sort the columns based on length of its cell values,

Reply:
Modified